pkg/controllers/pushsecret/pushsecret_controller.go

```diff
@@ -92,6 +92,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, fmt.Errorf("get resource: %w", err)
 	}
 
+	// skip when pointing to an unmanaged store
+	skip, err := shouldSkipUnmanagedStore(ctx, req.Namespace, r, ps)
+	if skip {
+		log.Info("skipping unmanaged store as it points to a unmanaged controllerClass")
```

Comment:
It would be better for this message to be clear on what secretStore is unmanaged, as we could have any number of SecretStores in the PushSecret (and only e.g. 2 of them not having the correct Controller Class)

Reply:
Since PS controller supports multiple SecretStores - I imagine we would still want to push secrets that have managed SecretStores and simply skip unmanaged ones.

I thought the best place to move the skip logic would be the [PushSecretToProviders](https://github.com/external-secrets/external-secrets/pull/3123/files/3eb66309850bf97e1da782db901501ee09a2fd24#diff-28773406fab3a6c16378745a672b49a6cf756f75db8f794c9ebb5886a44dd521R284) func, as we could also log the unmanaged stores here as per your comment. 

In these cases where we have at least 1 unmanaged SecretStore, what should the PushSecret status be at the end?

Reply:
Sorry for the very late reply - burning out on regular work 😄 For me :

> 0 unmanaged stores, >=1 managed stores - sync successfully

agree

> 1 unmanaged store, 0 managed stores - skip and PushSecretCondition is nil?

emit a Warning K8s Event as there are X unmanaged Stores (generate X events would probably be better here). Fail the PushSecret as there is no valid stores to push.

> 0 unmanaged stores, >0 managed stores - skip all unmanaged and push to all managed stores successfully?

emit Warning for the unmanaged Stores. Proceed normally for the managed ones.

Reply:
I have implemented these cases. However, for the case when all of the stores are unmanaged:
> emit a Warning K8s Event as there are X unmanaged Stores (generate X events would probably be better here). Fail the PushSecret as there is no valid stores to push.

Isn't the failing of the PushSecret the cause of the log spam? So this doesn't fix the original issue. In the cases where we have multiple ES controllers running with their own stores, should we not skip silently to avoid log spam from stores belonging to other controllers? (events too?) IIRC the logic for ES controllers is to skip silently too.

If we remove that `r.markAsFailed` I think that will remove the log spam https://github.com/external-secrets/external-secrets/pull/3123/files#diff-28773406fab3a6c16378745a672b49a6cf756f75db8f794c9ebb5886a44dd521R163

Example from when I deployed this to the cluster (deployed for ~10 mins):
```bash
✦ ❯ k events --for pushsecret/foo-ps
LAST SEEN              TYPE      REASON    OBJECT                         MESSAGE
27m (x12 over 27m)     Warning   Errored   PushSecret/foo-ps   PushSecret "foo-ps" has no managed stores to push to
22m (x1377 over 27m)   Warning   Errored   PushSecret/foo-ps   Controller for the store "devops" does not match the reconciler's controller "default", so the store "devops" is not managed by the controller "default"
```